### PR TITLE
Tutor Refactor

### PR DESCRIPTION
**Pull request by JDillon522**
I got it working and I also did a bit of cleanup to the code. There's still lots of finessing that could be done but the bulk of the original logic is in place. 

The main problem was that you gave multiple elements in the HTML the same `id` parameter. IDs have to be unique. So I changed all of them to classes. Then it was a simple matter of accessing the necessary one dynamically. 

Take a look at the code and we can comb through things and answer your questions next time we meet. Or you can just make comments on the lines where you have questions. Github will let me know :)

### CODE REVIEW

File: game.html [hunk 1]
```diff
@@ -1,379 +1,133 @@
...
-                    <span id="Pokervalue"></span><div id="suitcenter"><img src="" alt=""></div>
-                </div>
-                <div id="dcardspic">
-                    <div id="suitcorner"><img src="" alt=""></div>
-                    <span id="Pokervalue"></span><div id="suitcenter"><img src="" alt=""></div>
+        <h1>Welcome to Blackjack World</h1>
+        <article>
+            <section class="display-desktop">
+                <div>Dealer's Cards:
+                    <div class="cardsinfo">
```
Comment (sunnyyuanz) [inline game.html:59]: change id to class from line59 to 83

File: game.html [hunk 2]
```diff
@@ -1,379 +1,133 @@
...
-            }
-            
-        })
-
-        function BacktoGamestart(){
-            const url=window.location.href.replace("game.html","gamestart.html");
-            window.location.href=url;
-        };
+                <div>Player's Cards:
+                    <div class="cardsinfo">
```
Comment (sunnyyuanz) [inline game.html:90]: Change Id to class from line90 to 114

File: game.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
+
+//Pokercard only show when something in the logic breaks and Spade is the first suit.
+let Dealercardsinfo = document.getElementById("Dealercardsinfo");
+let Playercardsinfo = document.getElementById("Playercardsinfo");
+let Dealerstatus = document.getElementById("Dealerstatus");
+let Gamestatus = document.getElementById("Gamestatus");
+let hitbutton = document.getElementById("hit");
+let staybutton = document.getElementById("stay");
+let pcardspic = document.querySelectorAll(".pcardspic")
```
Comment (sunnyyuanz) [inline game.js:9]: instead of document.getelementbyid use document.querySelectorAll from line9 to 13

File: game.js [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+let dcardspic = document.querySelectorAll(".dcardspic")
+let suitTop = document.querySelectorAll(".suit-top")
+let suitBottom = document.querySelectorAll(".suit-bottom")
+let cardValue = document.querySelectorAll("card-value")
+let gameoverbutton = document.getElementById("gameover")
+const playerCards = [];
```
Comment (sunnyyuanz) [inline game.js:15]: add const in front of list instead of default let from line15 to 16

File: game.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+const dealerCards = [];
+let playerTotal = 0;
```
Comment (sunnyyuanz) [inline game.js:17]: Add let from line17 to 21

File: game.js [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+gameoverbutton.style.display = "none";
+
+
+
+const cards = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+let suits = ['Hearts', 'Clubs', 'Diamonds', 'Spades'];
+let values = ['A', 'K', 'Q', 'J', '10', '9', '8', '7', '6', '5', '4', '3', '2']
+let deck = createDeck()
+
+
```
Comment (sunnyyuanz) [inline game.js:32]: Removed Newgameshowcards function

File: game.js [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+        suit: suits[suitIdx],
+        value: values[valueIdx]
+      }
+      deck.push(card);
+    }
+  }
+  return deck;
+}
+
+function defineNum(card) {
```
Comment (sunnyyuanz) [inline game.js:47]: change card to tmp from line47 to 48

File: game.js [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+  switch (card.value) {
+    case 'A':
+      return 1;
+    case 'K':
+      return 13;
+    case 'Q':
+      return 12;
+    case 'J':
+      return 11;
+    default:
```
Comment (sunnyyuanz) [inline game.js:57]: Instead of listing out all cases, use default to claim the value for the rests since they are all just numbers

File: game.js [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+      return parseInt(card.value, 10);
+  }
+
+}
+
```
Comment (sunnyyuanz) [inline game.js:62]: removed console.log(cards.length); console.log(deck.length);

File: game.js [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+function getcardstring(tmp) {
+  return tmp.suit + tmp.value;
+}
+
+function ShowPoker(card, index, actor) {
+  const cardDom = document.querySelectorAll(actor)[index];
```
Comment (sunnyyuanz) [inline game.js:69]: Where is this actor in the document? What does line69 do? Assuming it's pulling in all element in Dom, then what does index do? Answer in line83 and 92

File: game.js [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+  cardDom.querySelector('.card-value').innerText = card.value
```
Comment (sunnyyuanz) [inline game.js:70]: update tmp to card from line70 to 73. And add a style to it so it's easier to adjust it in css later

File: game.js [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+  cardDom.querySelector('.suit-top').innerHTML = `<img src="images/${card.suit}.png"  alt="${card.suit}">`
+  cardDom.querySelector('.suit-bottom').innerHTML = `<img src="images/${card.suit}.png" alt="${card.suit}">`
+  cardDom.style.display = 'flex';
```
Comment (sunnyyuanz) [inline game.js:73]: Removed  return suitcorner.innerHTML
            return suitcenter.innerHTML
            return Pokervalue.innerText

File: game.js [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+function DrawNewDeckcards() {
+  for (dealerCardIndex = 0; dealerCardIndex < 2; dealerCardIndex++) {
+    let suit = 1 + Math.trunc(Math.random() * deck.length)
```
Comment (sunnyyuanz) [inline game.js:78]: change Dcard to suit, change tmp to card. Actor question is answered. Actor represents the parent classname that includes multiple children classnames.

File: game.js [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+    playerTotal += defineNum(card);
+    ShowPoker(card, playerCardIndex, '.pcardspic')
+  }
+}
+
+
+
+function Newgamedeck() {
+  DrawNewDeckcards()
+
```
Comment (sunnyyuanz) [inline game.js:100]: removed the 3 console.log

File: game.js [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+  Playercardsinfo.innerHTML = `<p id="Playercardsinfo">Player Cards: ${playerCards} -- Total: ${playerTotal}<p>`
+
+  checkgamestatus();
+
```
Comment (sunnyyuanz) [inline game.js:104]: Remove Newgameshowcards function and return values

File: game.js [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+  }
+
+}
+function DealerGetNewCards() {
+  if (dealerTotal <= 15) {
+    let Dcard = 1 + Math.trunc(Math.random() * deck.length);
+    let tmp = deck[Dcard]
+    dealerCards.push(getcardstring(tmp));
+    dealerTotal += defineNum(tmp);
+    ShowPoker(tmp, dealerCardIndex, '.dcardspic');
```
Comment (sunnyyuanz) [inline game.js:173]: add dealercardIndex,classname in ShowPoker(), Give dcardspic[dealerCardIndec] a style. 
add dealerCardIndex++
remove return

File: game.js [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+    Dealerstatus.innerText = "Dealer chose Stay"
+  }
+}
+
+hitbutton.addEventListener('click', function () {
+  let Pcard = 1 + Math.trunc(Math.random() * deck.length);
+  let tmp = deck[Pcard]
+  playerCards.push(getcardstring(tmp));
+  playerTotal += defineNum(tmp);
+  ShowPoker(tmp, playerCardIndex, '.pcardspic')
```
Comment (sunnyyuanz) [inline game.js:188]: Add ShowPoker()
Give pcard[playerIndec] a style
add playerCardIndec++

**Review by sunnyyuanz — COMMENTED** [review]
Thank you Joseph for sorting this out for me! I see that the major issues with my code was using Id for different section instead of class while Id must be unique for each section. Some unnecessary used of return in functions. Also learned use default in switch function to save lines of code. And how to use querySelectorAll in a smarter way. Give element style in the script. My question is Which funtion replaced Newgameshowcards function? And what does playerCardIndec++ and dealerCardIndex++ do exactly? And sorry, the comments in the middle of code are all for myself, not questions. Still getting familiar with GitHub. Lol

**Comment by JDillon522** [comment]
@sunnyyuanz no worries! :) Typically in a pull request, you leave a comment to discuss with others something, not so much taking notes. But hey, its your repo. You run it however you want :)

I honestly cant remember some of the logic right now. We can go over it a bit when we get together next

**Pull request merged**